client/modules/wj-state/src/i18n.ts

```diff
@@ -0,0 +1,140 @@
+// organize-imports-ignore
+import * as i18n from "svelte-i18n"
+import type { MessageFormatter } from "svelte-i18n/types/runtime/types"
+import type { Readable } from "svelte/store"
+
+// -- INIT
+
+// we use `en` as a fallback language, so rather than dynamically importing it
+// we will bundle it with the JS. this may be inefficient but it means that
+// a user with a spotty internet connection won't see every single string as garbage.
+// this should also prevent some edge case errors, hopefully
+import langEN from "../../../../locales/en.yaml"
```

Comment:
Do we really want this? It's kind of the hardcoded english issue, but for people with poor connections. Is there some alternative for say, french users?

Reply:
I can't think of a decent alternative, because the issue is that you'd need to serve a different JS file depending on locale which sounds messy.

Although there actually is a great alternative, using top level await, but that's a really modern language features and isn't supported everywhere yet. It means the JS execution would synchronously pause while the i18n code fetches the translation file. Also this would only work for languages where translation is complete, as otherwise the fallback might be missing while a non translated string is requested

Reply:
I don't really like how this privileges `en` as a language, vs our approach of the default debug language. While I don't like the idea users seeing debug strings, I'd rather all users see the same debug strings rather than non-english users seeing unexpected english.

Reply:
Unfortunately there is a difference between "the debug language" and synchronously loading `en`. In the mean time while no language is loaded, the translation service has no locales whatsoever and that did cause me some issues, for example if you call formatMessage too quickly it'll error out for a bizarre reason.

So we'd have to synchronously load an actual debug language and not just show the key names. Personally I'd rather just accept the EN fallback, at least if a string is untranslated you have something you can copy paste into Google Translate

Reply:
Well if you just copied all the key names that would work, but that's kind of a lot of kilobytes for ultimately still English text, just technical and bizarre looking English text.

Reply:
Right...

Oh well.